features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/resources/BitmapCachesManager.kt

```diff
@@ -51,15 +51,19 @@ internal class BitmapCachesManager(
         isBitmapPoolRegisteredForCallbacks = true
     }
 
-    internal fun putInResourceCache(drawable: Drawable, resourceId: String) {
-        resourcesLRUCache.put(drawable, resourceId.toByteArray(Charsets.UTF_8))
+    internal fun putInResourceCache(key: String, resourceId: String) {
+        resourcesLRUCache.put(key, resourceId.toByteArray(Charsets.UTF_8))
     }
 
-    internal fun getFromResourceCache(drawable: Drawable): String? {
-        val resourceId = resourcesLRUCache.get(drawable) ?: return null
+    internal fun getFromResourceCache(key: String): String? {
+        val resourceId = resourcesLRUCache.get(key) ?: return null
         return String(resourceId, Charsets.UTF_8)
     }
 
+    internal fun generateResourceKeyFromDrawable(drawable: Drawable): String? {
+        return (resourcesLRUCache as? ResourcesLRUCache)?.generateKeyFromDrawable(drawable)
```

Comment:
imo this opportunistic cast looks like a hack: if at some point the instance is not `ResourcesLRUCache`, then the flow will be malfunctioning, but it won't be any indication what is wrong exactly.

Reply:
I don't have a recommendation right now, because of my limited knowledge of the Android Session Replay components, but maybe create TODO item for the future refactoring?

Reply:
I thought the same ✅ I have added a TODO mentioning the ticket